docs/rust-client/custom_note_how_to.md

```diff
@@ -0,0 +1,443 @@
+* Leverage Miden’s privacy features to keep transaction details private.
+
+Unlike Ethereum, where transaction details are publicly visible in the mempool, Miden enables you to partially or completely hide transaction details.
+
+
+## What we'll cover
+
+- Creating custom note logic
+- Consuming notes
+
+## Step-by-Step Process
+
+### 1. Creating Two Accounts: Alice & Bob
+
+First, we create two basic accounts for the two users:
+- **Alice:** The account that creates and funds the custom note.
+- **Bob:** The account that will consume the note if he knows the secret.
+
+### 2. Deploying a Fungible Faucet
+
+Next, we deploy a fungible faucet. This faucet is used to mint assets (tokens) that will be included in the note. The asset is identified by a token symbol (e.g., MID) and minted to Alice’s account. This ensures that Alice has the necessary tokens to create the note.
+
+### 3. Hashing the Secret Number
+
+The security of the custom note hinges on a secret number. Here, we will:
+- Choose a secret number (for example, an array of four field elements).
+- For simplicity, we're only hashing 4 elements. Therefore, we prepend an empty word—consisting of 4 zero Felts—as a placeholder. This is required by the RPO hashing algorithm to ensure the input has the correct structure and length for proper processing.
+- Compute the hash of this secret. The resulting hash will serve as the note’s input, meaning that the note can only be consumed if the secret number’s hash preimage is provided during consumption.
+
+### 4. Creating the Custom Note
+
+Now, combine the minted asset and the secret hash to build your custom note. The note is created using the following key steps:
+
+1. **Note Inputs:**  
+   - The note is set up with the asset and the hash of the secret number.
+   
+2. **Miden Assembly Code:**  
+   - The custom Miden Assembly script ensures that the note can only be consumed if the provided secret, when hashed, matches the original hash stored in the note.
+
+Below is the Miden Assembly code for the note:
+
+```masm
+use.miden::note
+use.miden::contracts::wallets::basic->wallet
+
+# => [HASH_PREIMAGE_SECRET]
+begin
+
+    # => [HASH_PREIMAGE_SECRET]
+    hperm
+
+    # => [F,E,D]
+    # E is digest
+    dropw swapw dropw
+
+    # => [DIGEST]
+    # writing note inputs to memory
+    push.0 exec.note::get_inputs drop drop
+
+    # => [DIGEST]
+    # pad stack and load note inputs from memory
+    padw push.0 mem_loadw
+
+    # => [INPUTS, DIGEST]
+    # assert equality
+    assert_eqw
+
+    # => []
+    # write assets in note to memory address 0
+    push.0 exec.note::get_assets
+
+    # => [num_assets, dest_ptr]
+    drop
+
+    # => [dest_ptr]
+    # load asset from memory
+    mem_loadw
+
+    # => [ASSET]
+    # call receive asset in wallet
+    call.wallet::receive_asset
+
+end
+```
+
+### How the Assembly Code Works:
+
+1. **Passing the Secret:**  
+   The secret number is passed as an argument into the note.
```

Comment:
Can you refer here explicitly to `Note arguments` and maybe even link to the part in the docs?